pkg/rancher-desktop/backend/wsl.ts

```diff
@@ -1211,29 +1211,29 @@ export default class WSLBackend extends events.EventEmitter implements VMBackend
           this.progressTracker.action('Installing image scanner', 100, this.installTrivy()),
           this.progressTracker.action('Installing CA certificates', 100, this.installCACerts()),
           this.progressTracker.action('Installing helpers', 50, this.installWSLHelpers()),
-          this.progressTracker.action('Writing K3s configuration', 50, async() => {
-            const k3sConf = {
-              PORT:                   config.kubernetes.port.toString(),
-              LOG_DIR:                await this.wslify(paths.logs),
-              'export IPTABLES_MODE': 'legacy',
-              ENGINE:                 config.containerEngine.name,
-              ADDITIONAL_ARGS:        config.kubernetes.options.traefik ? '' : '--disable traefik',
-              USE_CRI_DOCKERD:        BackendHelper.requiresCRIDockerd(config.containerEngine.name, config.kubernetes.version).toString(),
-            };
-
-            if (!config.kubernetes.options.flannel) {
-              console.log(`Disabling flannel and network policy`);
-              k3sConf.ADDITIONAL_ARGS += ' --flannel-backend=none --disable-network-policy';
-            }
-
-            await this.writeConf('k3s', k3sConf);
-          }),
         ];
 
         if (kubernetesVersion) {
           const version = kubernetesVersion;
 
           installerActions.push(
+            this.progressTracker.action('Writing K3s configuration', 50, async() => {
+              const k3sConf = {
+                PORT:                   config.kubernetes.port.toString(),
+                LOG_DIR:                await this.wslify(paths.logs),
+                'export IPTABLES_MODE': 'legacy',
+                ENGINE:                 config.containerEngine.name,
+                ADDITIONAL_ARGS:        config.kubernetes.options.traefik ? '' : '--disable traefik',
+                USE_CRI_DOCKERD:        BackendHelper.requiresCRIDockerd(config.containerEngine.name, config.kubernetes.version).toString(),
```

Comment:
```suggestion
                USE_CRI_DOCKERD:        BackendHelper.requiresCRIDockerd(config.containerEngine.name, kubernetesVersion).toString(),
```

That means we don't accidentally have this issue anymore (because TypeScript would blow up because it's undefined).

Reply:
This seems "more correct" anyways, so we definitely need to do that. But TypeScript doesn't like it:

```
TS2345: Argument of type 'SemVer | undefined' is not assignable to parameter of type 'string | SemVer'.
Type 'undefined' is not assignable to type 'string | SemVer'.
```

I can just use `version` though, so will change to that.